demos/using-dev-build/vite.config.ts

```diff
@@ -0,0 +1,41 @@
+import { defineConfig, type UserConfig } from "vite";
+import { readCanisterIds } from "./vite.plugins";
+
+// npm run dev = local
+// npm run build = local
+// dfx deploy = local
+// dfx deploy --network ic = ic
+const network = process.env.DFX_NETWORK ?? ("local" as const);
+
+export default defineConfig(
+  ({ mode }: UserConfig): UserConfig => ({
+    root: "webapp",
+    build: {
+      outDir: "../dist",
+      emptyOutDir: true,
+      commonjsOptions: {
+        // Source: https://github.com/rollup/plugins/issues/1425#issuecomment-1465626736
+        strictRequires: true,
+      },
+    },
+    define: {
+      "process.env": {
+        ...readCanisterIds({ network }),
+        DFX_NETWORK: network,
```

Comment:
is this used?

Reply:
same as https://github.com/dfinity/internet-identity/pull/1485#discussion_r1176188911

Reply:
agent-js uses `DFX_NETWORK`. while we might not hits it, same as https://github.com/dfinity/internet-identity/pull/1485#discussion_r1176188911

Reply:
Ok gotcha; confusing still. Let's remove anything we don't need (also for the community's sake, otherwise other people might copy this blindly and also end up with unused values)